apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -247,13 +238,10 @@ def read(
             if value_filter is not None:
                 query_condition = QueryCondition(value_filter)
 
-            sr = clib.SOMAReader(
-                self._uri,
-                name=self.__class__.__name__,
+            sr = self._soma_reader(
```

Comment:
@mlin just a thought, to split this PR up & make it easier to iterate on, maybe have a PR that _only_ changes `sr = clib.SOMAReader` to `self._soma_reader` -- with no timestamps mods at all -- just a 'code motion' PR ...

Reply:
@mlin, I'm happy to take pass over the optimized nnz. It sounds like we are moving that to a separate PR.